packages/grid/_modules_/grid/components/editCell/EditInputCell.tsx

```diff
@@ -0,0 +1,77 @@
+import * as React from 'react';
+import InputBase, { InputBaseProps } from '@material-ui/core/InputBase';
+import { GridCellParams } from '../../models/params/gridCellParams';
+import { formatDateToLocalInputDate, isDate, mapColDefTypeToInputType } from '../../utils/utils';
+import { GridEditRowUpdate } from '../../models/gridEditRowModel';
+import { GridEditRowApi } from '../../models/api/gridEditRowApi';
+
+export function EditInputCell(props: GridCellParams & InputBaseProps) {
+  const {
+    value,
+    api,
+    field,
+    row,
+    colDef,
+    getValue,
+    rowIndex,
+    colIndex,
+    isEditable,
+    ...inputBaseProps
+  } = props;
+
+  const editRowApi = api as GridEditRowApi;
+  const [valueState, setValueState] = React.useState(value);
+
+  const onValueChange = React.useCallback(
+    (event) => {
+      const newValue = event.target.value;
+      const update: GridEditRowUpdate = {};
+      update[field] = {
+        value: colDef.type === 'date' || colDef.type === 'dateTime' ? new Date(newValue) : newValue,
+      };
+      setValueState(newValue);
+      editRowApi.setEditCellProps(row.id, update);
+    },
+    [editRowApi, colDef.type, field, row.id],
+  );
+
+  const onKeyDown = React.useCallback(
+    (event: React.KeyboardEvent) => {
+      if (!inputBaseProps.error && event.key === 'Enter') {
+        const update: GridEditRowUpdate = {};
+        update[field] = { value };
+        editRowApi.commitCellChange(row.id, update);
+      }
+
+      if (event.key === 'Escape') {
+        editRowApi.setCellMode(row.id, field, 'view');
+      }
+    },
+    [inputBaseProps.error, row.id, field, value, editRowApi],
+  );
+
+  const inputType = mapColDefTypeToInputType(colDef.type);
+  const formattedValue =
+    valueState && isDate(valueState)
+      ? formatDateToLocalInputDate({ value: valueState, withTime: colDef.type === 'dateTime' })
+      : valueState;
+
+  React.useEffect(() => {
+    // The value props takes priority over state in case it is overwritten externally. Then we override the state value.
+    setValueState((prev) => (value !== prev ? value : prev));
```

Comment:
As far as I can read this line, it's identical.
```suggestion
    setValueState(value);
```

Reply:
no if prop.value is different to the current state then update otherwise keep the state

Reply:
Which is exactly what my proposal is doing, no? :D 

> setValueState((prev) => (value !== prev ? value : prev));

<==>

> setValueState((prev) => (value === prev ? prev : value));

<==>

> setValueState((prev) => (value === prev ? value : value));

<==>

> setValueState((prev) => value);

<==>

> setValueState(value);

Reply:
my bad you're correct. I just wanted to make sure there was no extra rendering